probe-rs/src/probe/arm_debug_interface.rs

```diff
@@ -319,32 +312,46 @@ fn perform_swd_transfers<P: RawProtocolIo>(
 
     let result = probe.swd_io(io_sequence.direction_bits(), io_sequence.io_bits())?;
 
-    let mut read_index = 0;
+    let mut result_bits = &result[..];
 
     for (i, transfer) in transfers.iter_mut().enumerate() {
-        let response_direction = transfer.direction;
-        let additional_idle_cycles_after = transfer.idle_cycles_after;
-
-        let response = parse_swd_response(&result[read_index..], response_direction);
+        // There are two idle bits and eight request bits, the response comes directly after.
+        let response_offset = 2 + 8;
+        let response = parse_swd_response(&result_bits[response_offset..]);
 
         probe.probe_statistics().report_swd_response(&response);
 
-        tracing::debug!("Transfer result {}: {:x?}", i, response);
-
         transfer.status = match response {
-            Ok(val) => {
-                if transfer.direction == TransferDirection::Read {
-                    transfer.value = val;
-                }
+            Ok(response_bits) if transfer.direction == TransferDirection::Read => {
+                // Parity-check and extract the response
+                let parity_bit = response_bits[32] as u32;
```

Comment:
I do not understand how this is correct. I would expect the result_bits to contain the ack bits at which point the parity bit would be at index 35.

Maybe we should introduce a discrete response type to avoid this? :)

Reply:
`result_bits` does contain the ack bits, `response_bits` does not 👀

Reply:
Well, imo the ack bits are part of the response on the same level as the parity bit. And we check them on two different levels. Sounds wrong to me.

Reply:
Good point